src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneOfState = { ...state };
+  const oldVersionsArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneOfState, action.extraData);
+        oldVersionsArray.push({ ...cloneOfState });
```

Comment:
```suggestion
```
Don't repeat yourself. You can write this statement only once, at the end of the iteration.